addons/website_sale_delivery/controllers/main.py

```diff
@@ -13,10 +13,13 @@ class WebsiteSaleDelivery(WebsiteSale):
     def shop_payment(self, **post):
         order = request.website.sale_get_order()
         carrier_id = post.get('carrier_id')
+        keep_carrier = post.get('keep_carrier')
```

Comment:
I prefer if we default to False, avoiding None
```suggestion
        keep_carrier = post.get('keep_carrier', False)
```